rpc/src/eth.rs

```diff
@@ -804,12 +827,16 @@ impl<B, C, P, CT, BE, A> EthApiT for EthApi<B, C, P, CT, BE, A> where
 				return Ok(Vec::new());
 			}
 
-			let (block, _, statuses) = self.client.runtime_api()
-				.current_all(&id)
-				.map_err(|err| internal_err(format!("fetch runtime account basic failed: {:?}", err)))?;
+			let block: Option<ethereum::Block> = self.current_block(&id);
+			let statuses: Option<Vec<TransactionStatus>> = self.current_statuses(&id);
```

Comment:
You're replacing the call to `current_all` with individual calls. Doesn't that mean two calls into the runtime instead of one?

Reply:
Oh, or the entire point is that we're not using the runtime anymore. This is all calls to aux storage. If that's the case, why do we still have the stoage items in the runtime?

Reply:
This are not auxiliary calls but calls to the StateBackend (we are accessing and decoding the storage primitives).

Reply:
Okay, so we're still reading the data stored in the runtime, but we're reading it in a more "direct" way by calling through the StateBacked, right?

Then my original question remains: Why replace the call to `current_all`? isn't it faster?

Reply:
`current_all` belongs to the rpc runtime API impl. It's a single call to the runtime that loads 3 storage values from pallet-ethereum, but still a runtime call.